### PR TITLE
Integrate graphql-pundit

### PR DESCRIPTION
**Pull request by phyrog**
Part of #175. Authorizes some fields:

- UserType#email, UserType#public_keys, UserType#unconfirmed_email are authorized
- OrganizationalUnitType#repositories is scoped
- QueryType#repository is authorized

Also I added a test for authorizing the REST branches controller

### CODE REVIEW

File: app/policies/user_policy.rb [hunk 1]
```diff
@@ -11,6 +11,6 @@ def show_current_user?
   end
 
   def access_private_data?
-    current_user == resource
+    !!current_user && current_user.id == resource.id
```
Comment (eugenk) [inline app/policies/user_policy.rb]: Redundant `!!`:
`&&` will always return `true` or `false`.

Comment (phyrog) [reply app/policies/user_policy.rb]: Hm I had a problem that it returned `nil` instead of `false` when `current_user` was `nil`; will check again.

Comment (phyrog) [reply app/policies/user_policy.rb]:
```ruby
nil && 4 == 5
=> nil
```

Comment (eugenk) [reply app/policies/user_policy.rb]: Ah okay. Sorry for misleading you.

**Pull request merged**